src/pages/index.js

```diff
@@ -25,39 +28,55 @@ const cardSection = new Section(
 
 const profileEditButton = document.querySelector("#profile-edit-button");
 
-const handlePopupWithForm = new PopupWithForm("#profile-edit-modal", (data) => {
-  userInfo.setUserInfo({
-    name: data.title,
-    job: data.description,
-  });
-  handlePopupWithForm.close();
+const editProfilePopup = new PopupWithForm("#profile-edit-modal", (data) => {
+  editProfilePopup.renderModalFormLoading(true);
+  api
+    .editProfile(data.title, data.description)
+    .then((updatedUserInfo) => {
+      userInfo.setUserInfo({
+        name: updatedUserInfo.name,
+        job: updatedUserInfo.about,
+      });
+      editProfilePopup.close();
+    })
+    .catch((err) => console.error(err))
+    .finally(() => editProfilePopup.renderModalFormLoading(false));
 });
 
-handlePopupWithForm.setEventListeners();
+profileEditButton.addEventListener("click", () => {
+  const currentUserData = userInfo.getUserInfo();
+  editProfilePopup.setInputValues({ currentUserData });
+  editProfilePopup.open();
```

Comment:
Please, delete it


You have 2 profile opening click handlers in `index.js`. Please, delete the duplicate